profiles/management/commands/gen_realistic_users.py

```diff
@@ -0,0 +1,254 @@
+"""
+Generates a set of realistic users to help us test search functionality
+"""
+from datetime import datetime, timedelta
+from django.core.management import BaseCommand
+from django.contrib.auth.models import User
+from profiles.models import Employment, Education
+from courses.models import Program, Course, CourseRun
+from dashboard.models import CachedCertificate, CachedEnrollment
+from micromasters.utils import load_json_from_file
+
+
+USER_DATA_PATH = 'profiles/management/realistic_user_data.json'
+PROGRAM_DATA_PATH = 'profiles/management/realistic_program_data.json'
+FAKE_USER_USERNAME_PREFIX = 'fake.'
+FAKE_PROGRAM_DESC_PREFIX = '[FAKE] '
+CACHED_MODEL_LAST_REQUEST = datetime.now() + timedelta(days=365)
+
+
+# Util functions
+
+def first_matching_item(iterable, predicate):
+    """Returns the first item in an iterable that matches a given predicate"""
+    return next(x for x in iterable if predicate(x))
+
+
+def is_relation_field(model_obj, attr_name):
+    """
+    Checks if a given model attribute is a relation rather than a single value
+    """
+    return 'RelatedManager' in str(type(getattr(model_obj, attr_name)))
+
+
+def deserialize_model_data(model_obj, data, save=True):
+    """
+    Sets attributes on a model object from a dictionary if the keys are valid model attributes
+    """
+    for k, v in data.items():
+        if hasattr(model_obj, k) and not is_relation_field(model_obj, k):
+            setattr(model_obj, k, v)
+    if save:
+        model_obj.save()
+    return model_obj
+
+
+# User data deserialization
+
+class CachedModelDeserializer:
+    """
+    Base class for deserializing a model used for edX data caching (eg: Enrollments, Certificates)
+    """
+    model_cls = None
+
+    @classmethod
+    def _get_edx_course_key(cls, data):   # pylint: disable=unused-argument
+        """Gets the edx_course_key value from the data dictionary"""
+        return ''
+
+    @classmethod
+    def _fill_in_missing_data(cls, data, user, course_run):   # pylint: disable=unused-argument
+        """Fills in data using the associated User and CourseRun objects"""
+        return data
```

Comment:
these are supposed to be implemented by the child classes. i could do `raise NotImplementedError`, but i think i didn't do it because PyCharm and/or the linter were complaining and i just wanted it/them to shut up